src/ripple/app/consensus/RCLCxPeerPos.h

```diff
@@ -73,49 +70,38 @@ class RCLCxPeerPos
     Slice
     signature() const
     {
-        return data_->signature_;
+        return {signature_.data(), signature_.size()};
     }
 
     //! Public key of peer that sent the proposal
     PublicKey const&
     publicKey() const
     {
-        return data_->publicKey_;
+        return publicKey_;
     }
 
     //! Unique id used by hash router to suppress duplicates
     uint256 const&
     suppressionID() const
     {
-        return data_->suppression_;
+        return suppression_;
     }
 
     Proposal const&
     proposal() const
     {
-        return data_->proposal_;
+        return proposal_;
     }
 
     //! JSON representation of proposal
     Json::Value
     getJson() const;
 
 private:
-    struct Data : public CountedObject<Data>
-    {
-        PublicKey publicKey_;
-        Buffer signature_;
-        uint256 suppression_;
-        Proposal proposal_;
-
-        Data(
-            PublicKey const& publicKey,
-            Slice const& signature,
-            uint256 const& suppress,
-            Proposal&& proposal);
-    };
-
-    std::shared_ptr<Data> data_;
+    PublicKey publicKey_;
+    uint256 suppression_;
+    Proposal proposal_;
+    boost::container::static_vector<std::uint8_t, 72> signature_;
```

Comment:
I have a hypothetical question. If the size of the `signature_` variable was a multiple of 32 bytes, then
can we use a `ripple::base_uint<33>` for storing the `signature_` data? Will that be slower than `boost::static_vector`?

Is one of them faster than the other?

Reply:
`ripple::base_uint<33>` would not work as it specifies 33 bits. You need 72 bytes or 72 x 8 bits, and it is designed to store uint32_t not an array of bytes. Also `base_uint` doesn't manage a size, so you'd have to keep track separately of how long the signature is. `static_vector` is the right choice here!

Reply:
I see okay, thanks for the clarification.